src/include/migraphx/check_context.hpp

```diff
@@ -36,7 +38,23 @@ struct check_context
 {
     struct op : auto_register_op<op>
     {
-        std::string name() const { return "check_context::" + get_type_name<T>(); }
+        std::string name() const
+        {
+            const auto& op_type_name                      = get_type_name<T>();
+            const auto& split_name                        = split_string(op_type_name, ':');
+            std::vector<std::string> name_without_version = {"check_context"};
+            // op_type_name would contain internal namespace name with version_x_y_z
+            // remove version and construct op_name such as check_context::migraphx::gpu::context
+            std::for_each(split_name.begin(), split_name.end(), [&](const auto& i) {
```

Comment:
not sure how it can be done.

`copy_if` requires `destination` to be an iterator. and inside the predicate, I can't update the destination iterator.

Reply:
You can use `std::back_inserter` to insert into the vector.

```cpp
std::copy_if(split_name.begin(), split_name.end(),
             std::back_inserter(name_without_version), [&](const auto &i) {
               return not i.empty() and not contains(i, "version");
             });
return join_strings(name_without_version, "::");
```

Reply:
Yes, Thanks. 
Done.